dsm/resource_sobject.go

```diff
@@ -163,6 +169,16 @@ func contains(s []string, str string) bool {
 	return false
 }
 
+// this function takes a string in JSON format and unmarshals it. If the string is not in correct JSON format, it returns nil.
```

Comment:
I don't think this function does anything anymore

Reply:
It just unmarshals a string and returns the value. If there is any error then it returns the error.

Reply:
So then why not just unmarshal the string in situ?

Reply:
If I directly call the unmarshal function, then I have to add below two lines to the code.

**type mapFormat map[string]interface{}
var inputMap mapFormat**

Unmarshal function has been used twice during create and update. If I don't use the bothfunction the above code will be repeated. And in future if any new attribute is added (for example "dsa") then this function can be reused.

Reply:
I'm gonna say that's another reason to dislike Go, so I guess that's fine.